jenkins-job-builder/integration-jobs.yml

```diff
@@ -602,10 +687,61 @@
                     name: ipa.test
                     type: IPA
 
-        - '{prefix}-timed-integration-{project}-{pretty_name}-domlevel-{domain_level}':
-            pretty_name: replica_promotion
+        - '{prefix}-integration-{project}-{pretty_name}-domlevel-{domain_level}':
+            pretty_name: replica_promotion_install
+            suite: test_integration/test_replica_promotion.py
+            controller_suffix: ''
+            filter: '"Install"'
+            config_template: |
+                domain_level: {domain_level}
+                domains:
+                  - hosts:
+                      - name: master.ipa.test
+                        role: master
+                      - name: replica.ipa.test
+                        role: replica
+                      - name: replica.ipa.test
+                        role: replica
+                    name: ipa.test
+                    type: IPA
+
+        - '{prefix}-integration-{project}-{pretty_name}-domlevel-{domain_level}':
+            pretty_name: replica_promotion_promotion
+            suite: test_integration/test_replica_promotion.py
+            controller_suffix: ''
+            filter: '"Promotion"'
+            config_template: |
+                domain_level: {domain_level}
+                domains:
+                  - hosts:
+                      - name: master.ipa.test
+                        role: master
+                      - name: replica.ipa.test
+                        role: replica
+                    name: ipa.test
+                    type: IPA
+
+        - '{prefix}-integration-{project}-{pretty_name}-domlevel-{domain_level}':
+            pretty_name: replica_promotion_old_replica
+            suite: test_integration/test_replica_promotion.py
+            controller_suffix: ''
+            filter: '"Unprivileged or WrongClient or OldReplica"'
+            config_template: |
+                domain_level: {domain_level}
+                domains:
+                  - hosts:
+                      - name: master.ipa.test
+                        role: master
+                      - name: replica.ipa.test
+                        role: replica
+                    name: ipa.test
+                    type: IPA
+
+        - '{prefix}-integration-{project}-{pretty_name}-domlevel-{domain_level}':
+            pretty_name: replica_promotion_catch_all
             suite: test_integration/test_replica_promotion.py
             controller_suffix: ''
+            filter: '"not (Promotion or Install or Unprivileged or WrongClient or Old)Replica"'
```

Comment:
What does this filter mean? Is it doing concatenation of every operand in the expression?